nomos-services/key-management-system/src/secure_key.rs

```diff
@@ -2,6 +2,6 @@ use bytes::Bytes;
 use overwatch_rs::DynError;
 
 pub trait SecuredKey {
-    fn sign(&self) -> Result<Bytes, DynError>;
+    fn sign(&mut self, data: Bytes) -> Result<Bytes, DynError>;
```

Comment:
Does it make sense for a `sign` to need `mut self`?

Reply:
I think it is because some keys use a random state inside that need to advance. Can you confirm @youngjoon-lee ?

Reply:
It's because I used [`ed25519_dalek::ed25519::signature::SignerMut`](https://docs.rs/signature/2.2.0/signature/trait.SignerMut.html) instead of [`ed25519_dalek::ed25519::signature::Signer`](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/trait.Signer.html#method.sign), but actually we don't need to, right now. 
I removed the `mut`. Thank you.
3d2156a5